header/header_test.go

```diff
@@ -36,3 +39,33 @@ func TestMakeExtendedHeaderForEmptyBlock(t *testing.T) {
 
 	assert.Equal(t, EmptyDAH(), *headerExt.DAH)
 }
+
+func TestMismatchedDataHash_ComputedRoot(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	t.Cleanup(cancel)
+
+	_, client := core.StartTestCoreWithApp(t)
+	fetcher := core.NewBlockFetcher(client)
+
+	store := mdutils.Bserv()
+
+	sub, err := fetcher.SubscribeNewBlockEvent(ctx)
+	require.NoError(t, err)
+	<-sub
+
+	height := int64(1)
+	b, err := fetcher.GetBlock(ctx, &height)
+	require.NoError(t, err)
+
+	comm, val, err := fetcher.GetBlockInfo(ctx, &height)
+	require.NoError(t, err)
```

Comment:
simplification: If the goal is to unit test the new check, then all this setup is not necessary. We can use TestSute which is simpler and generates a valid header and we can similarly augment it with an incorrect hash to see that error happens.

Reply:
done.